mesonbuild/environment.py

```diff
@@ -1148,6 +1157,17 @@ def detect_static_linker(self, compiler):
         self._handle_exceptions(popen_exceptions, linkers, 'linker')
         raise EnvironmentException('Unknown static linker "%s"' % ' '.join(linkers))
 
+    def get_install_path(self, name: str) -> str:
+        """Helper that checks the native file as well as command line options,
+        and defaults and resolve precidence.
+        """
+        cd = self.coredata.builtins[name]
+        if cd.value_is_default():
+            conf = getattr(self.paths[MachineChoice.is_native(not self.is_cross_build())], name)
```

Comment:
See above, should be `self.paths.host`

Reply:
That wouldn't be correct as the code is right now, right now the native file is always read into the build attribute, and the cross file is always read into the host attribute.

Reply:
Ah,okay, yes. it took me a bit to convince myself that calling default_missing in this case was acutally safe, but yes.

Reply:
:) Glad you are convinced! It's a weird thing seemingly denormalizing everything up front, but since it avoids everything else being conditional (and since they're immutable thereafter so they cannot drift apart), I think it's the right decision.